k9mail-library/src/test/java/com/fsck/k9/mail/transport/smtp/SmtpTransportTest.java

```diff
@@ -787,6 +793,38 @@ public void sendMessagePipelining_without354ReplyforData_shouldThrow() throws Ex
         server.verifyInteractionCompleted();
     }
 
+    @Test
+    public void sendMessagePipelining_with250and550ReplyforRecipients_shouldThrow() throws Exception {
+        Message message = getMessageWithTwoRecipients();
+        MockSmtpServer server = createServerAndSetupForPlainAuthentication("PIPELINING");
+        server.expect("MAIL FROM:<user@localhost>");
+        server.expect("RCPT TO:<user2@localhost>");
+        server.expect("RCPT TO:<user3@localhost>");
+        server.expect("DATA");
+        server.output("250 OK");
+        server.output("250 OK");
+        server.output("550 remote mail to <user3@localhost> not allowed");
+        server.output("354 End data with <CR><LF>.<CR><LF>");
+        server.expect(".");
+        server.output("554 no valid recipients given");
```

Comment:
We are throwing an exception when we receiving a negative reply for recipient which shows that message was not send .
We do not care what server reply after the empty message because in any case(success or failure) we will throw exception (because of above) .

So I think it does not matter what we test for response of  empty message .